providers/providers_manager.go

```diff
@@ -269,7 +284,7 @@ func (pm *ProviderManager) GetProviders(ctx context.Context, k []byte) []peer.ID
 	case <-ctx.Done():
 		return nil
 	case peers := <-gp.resp:
-		return peers
+		return peerstore_legacy.PeerInfos(pm.pstore, peers)
```

Comment:
I think you can use peerstore.AddrInfos here, no?

Reply:
It's only present in go-libp2p-core v0.9.0 and up, however bumping to this release introduces a ton of other problems. Many of the libraries that this repo depends on, as well as the tests in the repo, are incompatible with v0.9.0 and up.

Reply:
hmmm... ok I guess. We're going to be using the later go-libp2p-core anyway in places like go-ipfs, but we can leave this for now.

In that case I wouldn't call the other one peerstore_legacy, it's where the common peerstore implementations live and definitely isn't legacy. Perhaps peerstore_impl or something like that.